### PR TITLE
Improvement / HOLO 1258 - Holograph Error

### PR DESCRIPTION
**Pull request by BravoNatalie**
## Describe Changes

Changes made to the package: **sdk**.

I made it better by doing ...

- Add `HolographError` class

## Checklist before requesting a review

- [x] I have performed a self-review of my code
- [x] Code styles have been enforced
- [x] All tests are passing

### CODE REVIEW

File: packages/sdk/src/contracts/holograph.contract.ts [hunk 1]
```diff
@@ -6,6 +6,11 @@ import {Config} from '../services/config.service'
 import {HolographABI} from '../constants/abi/develop'
 import {HolographByNetworksResponse, getContract, getSelectedNetworks, mapReturnType} from '../utils/contracts'
 import {HolographLogger} from '../services/logger.service'
+import {normalizeException} from '../errors/error-info'
+import {isCallException} from 'ethers'
```
Comment (feliperodriguess) [inline packages/sdk/src/contracts/holograph.contract.ts:10]: move this import to the top, to keep the pattern of package imports first, then file imports

File: packages/sdk/src/errors/error-info.ts [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+  return err
+}
+
+export const errorInfo = {
+  [HolographErrorCode.HOLO_SDK_10000]: () => ({
+    message: `Unknown error.`,
+    description: 'An error of unknown source was identified',
+  }),
+  [HolographErrorCode.HOLO_SDK_10001]: (chainId: BigNumberish) => ({
+    message: `Failed to get network to chainId ${chainId}.`,
```
Comment (feliperodriguess) [inline packages/sdk/src/errors/error-info.ts:23]: maybe the word `for` would be better, not sure tho, like:

`Failed to get network for chainId ${chainId}.`

File: packages/sdk/src/errors/general/unavailable-network.error.ts [hunk 3]
```diff
@@ -0,0 +1,18 @@
+import {BigNumberish} from 'ethers'
+import {errorInfo, HolographErrorCode} from '../error-info'
```
Comment (feliperodriguess) [inline packages/sdk/src/errors/general/unavailable-network.error.ts:2]: break line here, to separate the package import above

File: packages/sdk/src/errors/error-info.ts [hunk 4]
```diff
@@ -0,0 +1,34 @@
+import {BigNumberish} from 'ethers'
+
+export enum HolographErrorCode {
```
Comment (alexanderattar) [inline packages/sdk/src/errors/error-info.ts:3]: Are we sure error codes is really the best pattern? Generally, I am not a fan as error codes don't really have any inherent meaning. I think a pattern that uses named errors would be more intuitive and preferable, but that's just me

File: packages/sdk/src/errors/error-info.ts (hunk 2 above)
Comment (sogoiii) [reply packages/sdk/src/errors/error-info.ts:23]: @feliperodriguess you are right, it should be `Failed to get network for chain id ${chainId}`

**Review by sogoiii — CHANGES_REQUESTED** [review]

File: packages/sdk/src/contracts/registry.contract.ts [hunk 5]
```diff
@@ -7,6 +7,8 @@ import {HolographByNetworksResponse, getContract, getSelectedNetworks, mapReturn
 import {Address} from 'abitype'
 import {Providers} from '../services'
 import {HolographLogger} from '../services/logger.service'
+import {ContractRevertError, EthersError, HolographError} from '../errors'
+import {isCallException} from 'ethers'
```
Comment (feliperodriguess) [inline packages/sdk/src/contracts/registry.contract.ts:11]: move this to the top

**Review by feliperodriguess — APPROVED** [review]

**Review by sogoiii — APPROVED** [review]
👍

**Pull request merged**